# HG changeset patch
# User Falk Tannhäuser <falk.tannhauser@free.fr>
# Date 1376703409 25200
#      Fri Aug 16 18:36:49 2013 -0700
# Node ID 53f1d5bb58bcdbce7424afcf893ed55c0866e138
# Parent  aa09b746fa16b6bdb4c9f0f9316be6dc2a047bc7
__plt__.m: Fix legend when plotting vector against matrix (bug #39542)

* doc/interpreter/contributors.in: Add Falk Tannhäuser to list of contributors.

* scripts/plot/private/__plt__.m: Use for loop index i to index list of legend
strings.

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -268,16 +268,17 @@ Russell Standish
 Brett Stewart
 Doug Stewart
 Jonathan Stickel
 Judd Storrs
 Thomas Stuart
 Ivan Sutoris
 John Swensen
 Daisuke Takago
+Falk Tannhäuser
 Ariel Tankus
 Matthew Tenny
 Georg Thimm
 Duncan Temple Lang
 Kris Thielemans
 Olaf Till
 Christophe Tournery
 Thomas Treichl
diff --git a/scripts/plot/private/__plt__.m b/scripts/plot/private/__plt__.m
--- a/scripts/plot/private/__plt__.m
+++ b/scripts/plot/private/__plt__.m
@@ -142,17 +142,17 @@ endfunction
 
 function [hlgnd, tlgnd, setlgnd] = __plt_key__ (h, options, hlgnd, tlgnd, setlgnd)
   n = numel (h);
   if (numel (options) == 1)
     options = repmat (options(:), n, 1);
   endif
 
   for i = 1 : n
-    key = options.key;
+    key = options(i).key;
     if (! isempty (key))
       hlgnd = [hlgnd(:); h(i)];
       tlgnd = {tlgnd{:}, key};
       setlgnd = true;
     endif
   endfor
 endfunction
 
